logic/strategy/card_utils_test.go

```diff
@@ -0,0 +1,125 @@
+package strategy
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+)
+
+func validateHand(origHand, thisHand []model.Card) bool {
+	prev := make(map[model.Card]int)
```

Comment:
🌵 `thisHandMap`. and we can declare the size of it: `make(map[model.Card]struct{}, len(thisHand))`

also; golang does this thing where `map[/*anything*/]struct{}` takes up no memory. It's crazy and it shouldn't work, but it doesn't reserve memory for the empty structs and it abstracts the memory space for the keys away in the golang craziness somewhere.

tl;dr use `map[/*type*/]struct{}` whenever you don't care about the value, just the existence